lib/ravelin/authentication_mechanisms/magic_link.rb

```diff
@@ -0,0 +1,27 @@
+module Ravelin
+  module AuthenticationMechanisms
+    class MagicLink < RavelinObject
+      DELIVERING_METHODS = %w(email sms)
+      FAILURE_REASONS = %w(INVALID_LINK TIMEOUT INTERNAL_ERROR RATE_LIMIT BANNED_USER)
+
+      attr_accessor :transport, :success, :email, :phone_number, :failure_reason
+      attr_required :transport, :success
+
+      def failure_reason=(reason)
+        @failure_reason = reason.to_s.upcase
+      end
+
+      def validate
```

Comment:
We need to validate the presence of `phoneNumber` if `transport == "sms"` and the presence of `email` if `transport == "email"`, to make sure we comply with the API definition

Reply:
Added, thanks